File: 2-javascript/intermediate/missingLetters.js
/*jslint node: true */
'use strict';

var test = 'abcdefghijklmnopqrstuvwxyz';

function fearNotLetter (str) {
  var begin = test.indexOf(str[0]);
  console.log(test[begin]);
  for (var i = 0; i < str.length; i++) {
    if (str[i] !== test[begin + i]) {
      console.log('BINGO --> ', str[i], ' !== ', test[begin + i]);
      return test[begin + i];
    }
  }
  return undefined;
}

console.log('d --> ', fearNotLetter('abce'));
console.log('i --> ', fearNotLetter('abcdefghjklmno'));
console.log('undefined --> ', fearNotLetter('bcd'));
console.log('undefined --> ', fearNotLetter('yz'));

File: old-v1/goCode/intermediateGo/missingLetters.go
package main

import (
	"fmt"
	"strings"
)

func fearNotLetter(str string) string {
	var missing = ""
	letters := "abcdefghijklmnopqrstuvwxyz"
	test := strings.Split(letters, "")
	//fmt.Println(len(test))
	begin := strings.Index(letters, string(str[0]))
	//fmt.Println(begin)
	for i := 0; i < len(str); i++ {
		//fmt.Println("Compare ...", string(str[i]), " con ", test[begin+i])
		if string(str[i]) != test[begin+i] {
			missing = string(test[begin+i])
			return "On " + str + " Fails ... " + missing
		}
	}
	return "On " + str + " nobody fails"
}

func missingLetters() {
	fmt.Println(fearNotLetter("bcdf"))
	fmt.Println(fearNotLetter("abce"))
	fmt.Println(fearNotLetter("abcdefghjklmno"))
	fmt.Println(fearNotLetter("bcd"))
	fmt.Println(fearNotLetter("yz"))
}
